packages/edit-site/src/components/page-pages/index.js

```diff
@@ -150,24 +164,36 @@ export default function PagePages() {
 				enableSorting: false,
 			},
 		],
-		[ postStatuses ]
+		[ postStatuses, view.type ]
 	);
 
 	const trashPostAction = useTrashPostAction();
 	const actions = useMemo( () => [ trashPostAction ], [ trashPostAction ] );
 
 	// TODO: we need to handle properly `data={ data || EMPTY_ARRAY }` for when `isLoading`.
 	return (
-		<Page title={ __( 'Pages' ) }>
-			<DataViews
-				paginationInfo={ paginationInfo }
-				fields={ fields }
-				actions={ actions }
-				data={ pages || EMPTY_ARRAY }
-				isLoading={ isLoadingPages }
-				view={ view }
-				onChangeView={ setView }
-			/>
-		</Page>
+		<>
+			<Page title={ __( 'Pages' ) }>
+				<DataViews
+					paginationInfo={ paginationInfo }
+					fields={ fields }
+					actions={ actions }
+					data={ pages || EMPTY_ARRAY }
+					isLoading={ isLoadingPages }
+					view={ view }
+					onChangeView={ setView }
+				/>
+			</Page>
+			{ view.type === 'side-by-side' && (
+				<Page>
+					<div className="edit-site-page-pages-preview">
+						<Editor
+							postId={ view.selectedPostId }
+							postType={ view.selectedPostType }
+						/>
+					</div>
+				</Page>
+			) }
```

Comment:
Ideally this is part of the view component. Maybe a slot?

Reply:
Because it needs to be rendered into a separate Page component?

Reply:
Ok, so for me, it's either too early for this particular view to be implemented or we need to redesign it a bit. In the sense that instead of it being a separate page/frame, I think it should be within the same frame and just separated from the list with a light border: most applications that use this pattern do this (Mac OS notes app is the simplest example).

If it needs to breakout of the "Page" component, I think it introduces too much complexity that we don't need at this stage of development for a small outcome. It starts to raise the question of whether it's a "layout" or something else more global. (The layout being just the left sidebar)

cc @SaxonF

Reply:
I disagree with it being a small outcome

Reply:
The space in between is just styling at the end of the day. The bigger reason it's separate is because there should be no "Pages" heading at the top.

Reply:
I understand Riad's concern. I too was looking at the whole picture in terms of:

- A consumer, like PagePages, provides view settings, actions, and a bunch of entity records.
- The rest is mostly left to DataViews: mostly rendering lists, but potentially rendering some details (for now) and doing other handling on the entity records.
- The consumer mostly reacts to changes in the high-level data (e.g. change query args for the entities).

I think the line between what is abstracted by DataViews and what is up for the consumer to handle is a line likely to change as we progress, but for now I also feel we should intentionally restrict our model. Ella, do you feel it's incompatible to have the Editor inside ViewSideBySide?

Reply:
@mcsf I don't know how you see that work? This is the view component area? How do you fit in a preview?

<img width="1352" alt="image" src="https://github.com/WordPress/gutenberg/assets/4710635/a4c7b9ff-5121-412c-82f9-0cd4f79d23af">

Reply:
Other than a slot, not sure what solutions you all have in mind?